pgrollup/parser_pg.py

```diff
@@ -0,0 +1,290 @@
+    null = lambda self, _: None
+    true = lambda self, _: True
+    false = lambda self, _: False
+
+parser = Lark(grammar, start='value', lexer='standard')
+
+import pprint
+def parse(text):
+    '''
+    >>> pprint.pprint(parse(sql0))
+    '''
+    tree = parser.parse(text)
+    return Transformer().transform(tree)
+
+def parse_tree(text):
+    print(parser.parse(text).pretty())
+
+################################################################################
+# internal helper functions
+################################################################################
+
+def expr_to_str(expr):
+    '''
+    This function tramsforms expressions into strings.
+    
+    Example input:
+    {'TYPE': 'AEXPR',
+    'lexpr': {'TYPE': 'COLUMNREF', 'fields': ['test'], 'location': 80},
+    'location': 84,
+    'name': ['>='],
+    'rexpr': {'TYPE': 'COLUMNREF', 'fields': ['from'], 'location': 86}}
+    Exmpale output:
+    'test>=from'
+    '''
+    if expr['TYPE']=='A_CONST':
+        return str(expr['val'])
+    elif expr['TYPE']=='COLUMNREF':
+        return str(expr['fields'][0])
+    elif expr['TYPE']=='FUNCCALL':
+        if expr['agg_star']==True:
+            return expr['funcname'][0]+'(*)'
+        else:
+            return expr['funcname'][0]+'('+expr_to_str(expr['args'][0])+')'
+#           else:
+#               return expr['funcname'][0]+'('+expr['args'][0]['fields'][0]+')'
+    elif expr['TYPE']=='AEXPR':
+        return '('+expr_to_str(expr['lexpr']) + expr['name'][0] + expr_to_str(expr['rexpr'])+')'
+
+def _gettype(info):
+    '''
+    This function returns a query's join type
+    '''
+    if info['jointype'] == 0:
+        return "INNER JOIN"
+    elif info['jointype'] ==1:
+        return "OUTER JOIN"
+    elif info['jointype'] ==2:
+        return "FULL JOIN"
+
+def _getalias(info):
+    '''
+    This function returns any aliases for a selected column
+    '''
+    if info['name'] != None:
+        return info['name'][1]
+    else:
+        return ''
+
+def _getgroups(sql):
+
+    if sql[0]['stmt']['groupClause'] != None:
+        return [[x['fields'][0],str(x['fields'][0])] for x in sql[0]['stmt']['groupClause']]
+    else:
+        return ''
+def _getjoins(info):
+    '''
+    This function generates a list that contains join information
+    '''
+    join_infos = []
+    if info['TYPE']=='RANGEVAR':
+        table_name = str(info['relname'][1])
+        if info['alias']!= None:
+            table_alias=str(info['alias']['aliasname'][1])
+        else:    
+            table_alias = str(info['relname'][1])
+        join_type='FROM'
+        condition= ''
+        join_info = {
+            'table_name': table_name,
+            'table_alias': table_alias,
+            'condition': condition,
+            'join_type': join_type
+       }
+        join_infos.append(join_info)
+        return join_infos
+    elif info['TYPE']=='JOINEXPR':
+#       table_name = str(info['larg']['relname'][1])
```

Comment:
it is never acceptable to have commented out code in a pull request;

remember that the code is first and foremost intended to be read by humans, and these comments just get in the way